tests/e2e/e2e_gov_test.go

```diff
@@ -0,0 +1,296 @@
+*/
+func (s *IntegrationTestSuite) SendTokensFromNewGovAccount() {
+	s.writeGovProposals(s.chainA)
+	chainAAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainA.id][0].GetHostPort("1317/tcp"))
+	senderAddress, err := s.chainA.validators[0].keyInfo.GetAddress()
+	s.Require().NoError(err)
+	sender := senderAddress.String()
+	// Gov tests may be run in arbitrary order, each test must increment proposalCounter to have the correct proposal id to submit and query
+	proposalCounter++
+	s.T().Logf("Proposal number: %d", proposalCounter)
+
+	s.fundCommunityPool(chainAAPIEndpoint, sender)
+
+	s.T().Logf("Submitting Legacy Gov Proposal: Community Spend Funding Gov Module")
+	s.submitLegacyGovProposal(chainAAPIEndpoint, sender, fees.String(), "community-pool-spend", proposalCounter, configFile("proposal.json"))
+	s.T().Logf("Depositing Legacy Gov Proposal: Community Spend Funding Gov Module")
+	s.depositGovProposal(chainAAPIEndpoint, sender, fees.String(), proposalCounter)
+	s.T().Logf("Voting Legacy Gov Proposal: Community Spend Funding Gov Module")
+	s.voteGovProposal(chainAAPIEndpoint, sender, fees.String(), proposalCounter, "yes", false)
+
+	initialGovBalance, err := getSpecificBalance(chainAAPIEndpoint, govModuleAddress, uatomDenom)
+	s.Require().NoError(err)
+	proposalCounter++
+
+	s.T().Logf("Submitting Gov Proposal: Sending Tokens from Gov Module to Recipient")
+	s.submitNewGovProposal(chainAAPIEndpoint, sender, proposalCounter, configFile("proposal_2.json"))
+	s.T().Logf("Depositing Gov Proposal: Sending Tokens from Gov Module to Recipient")
+	s.depositGovProposal(chainAAPIEndpoint, sender, fees.String(), proposalCounter)
+	s.T().Logf("Voting Gov Proposal: Sending Tokens from Gov Module to Recipient")
+	s.voteGovProposal(chainAAPIEndpoint, sender, fees.String(), proposalCounter, "yes", false)
+	s.Require().Eventually(
+		func() bool {
+			newGovBalance, err := getSpecificBalance(chainAAPIEndpoint, govModuleAddress, uatomDenom)
+			s.Require().NoError(err)
+
+			recipientBalance, err := getSpecificBalance(chainAAPIEndpoint, govSendMsgRecipientAddress, uatomDenom)
+			s.Require().NoError(err)
+			return newGovBalance.IsEqual(initialGovBalance.Sub(sendGovAmount)) && recipientBalance.Equal(initialGovBalance.Sub(newGovBalance))
+		},
+		15*time.Second,
+		5*time.Second,
+	)
+}
+
+/*
+GovSoftwareUpgrade tests passing a gov proposal to upgrade the chain at a given height.
+Test Benchmarks:
+1. Submission, deposit and vote of message based proposal to upgrade the chain at a height (current height + buffer)
+2. Validation that chain halted at upgrade height
+3. Teardown & restart chains
+4. Reset proposalCounter so subsequent tests have the correct last effective proposal id for chainA
+TODO: Perform upgrade in place of chain restart
+*/
+func (s *IntegrationTestSuite) GovSoftwareUpgrade() {
+	chainAAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainA.id][0].GetHostPort("1317/tcp"))
```

Comment:
can we use a subtest?

```suggestion
	s.Run("description.....", func() {
		chainAAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainA.id][0].GetHostPort("1317/tcp"))
		...
```